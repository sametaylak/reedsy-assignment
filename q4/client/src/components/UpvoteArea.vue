<template>
  <div class="upvote-area">
    <span>Upvoted {{ book.upvotes }} times</span>
    <button :class="{ upvoted: book.upvoted, upvote: !book.upvoted }">
      {{ book.upvoted ? 'Upvoted' : 'Upvote' }}
    </button>
  </div>
</template>

<script>
export default {
  name: 'UpvoteArea',
  props: {
    book: Object
  }
}
</script>

<style lang="scss" scoped>
.upvote-area {
  span {
    margin-right: 16px;
  }
  button {
    border-radius: 4px;
    outline: none;
    cursor: pointer;
    padding: 4px 8px;
    font-size: 14px;
    &.upvote {
      border: none;
      color: white;
      background-color: #ebc02a; 
    }
    &.upvoted {
      border: 1px solid #ebc02a;
      color: #ebc02a;
    }
  }
}
</style>
