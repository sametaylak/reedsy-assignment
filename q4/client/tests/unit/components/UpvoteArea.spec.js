import { shallowMount } from '@vue/test-utils'
import UpvoteArea from '@/components/UpvoteArea.vue'

describe('UpvoteArea.vue', () => {
  it('has a book prop', () => {
    const wrapper = shallowMount(UpvoteArea)

    expect(wrapper.props().book).toBeUndefined();
  })

  it('button text must be upvoted and has a upvoted class', () => {
    const book = {
      author: 'Samet A.',
      title: 'Odyssey',
      slug: 'odyssey',
      synopsis: 'best book ever',
      upvoted: true,
      upvotes: 101
    }

    const wrapper = shallowMount(UpvoteArea, {
      propsData: { book }
    })

    expect(wrapper.props().book).toEqual(book)
    expect(wrapper.find('button.upvoted').exists()).toBe(true)
    expect(wrapper.find('button.upvoted').text()).toEqual('Upvoted')
    expect(wrapper.find('button.upvote').exists()).toBe(false)
  })

  it('button text must be upvote and has a upvote class', () => {
    const book = {
      author: 'Samet A.',
      title: 'Odyssey',
      slug: 'odyssey',
      synopsis: 'best book ever',
      upvoted: false,
      upvotes: 101
    }

    const wrapper = shallowMount(UpvoteArea, {
      propsData: { book }
    })

    expect(wrapper.props().book).toEqual(book)
    expect(wrapper.find('button.upvote').exists()).toBe(true)
    expect(wrapper.find('button.upvote').text()).toEqual('Upvote')
    expect(wrapper.find('button.upvoted').exists()).toBe(false)
  })
})
