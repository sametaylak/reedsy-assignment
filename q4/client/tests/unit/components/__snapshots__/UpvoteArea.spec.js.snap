// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpvoteArea.vue should have consistent dom 1`] = `
<div class="upvote-area"><span>Upvoted 101 times</span> <button class="upvote">
    Upvote
  </button></div>
`;
